ocfweb/component/banner.py

```diff
@@ -0,0 +1,23 @@
+from collections import namedtuple
+from urllib.request import urlopen
+
+import yaml
+
+
+class BannerMessage(namedtuple('BannerMessage', [
+    'visible',
+    'force_lab_closed',
+    'html',
+])):
+    pass
+
+
+def get_banner_message():
+    """Get the front page banner message from the default location."""
+    with urlopen('file:///home/s/st/staff/banner.yaml') as f:
```

Comment:
it'd be nice if those could work without NFS, or at least not fail loudly. Often people are running these on staff VMs which won't have this file. Usually we symlink the file into `~staff/public_html` and write code like:

```python
def get_vhost_db():
    """Returns lines from the vhost database. Loaded from the filesystem (if
    available), or from the web if not."""
    try:
        with open('/home/s/st/staff/vhost/vhost.conf') as f:
            return list(map(str.strip, f))
    except IOError:
        # fall back to database loaded from web
        return requests.get(constants.VHOST_DB_URL).text.split('\n')
```

Just make sure to chown the symlink (you'll have to use `-h` to not dereference it) to `staff:staff`.